crates/dslab-mp/tests/test_mc.rs

```diff
@@ -360,29 +458,51 @@ fn two_nodes_started_trace() -> Vec<LogEntry> {
     ]
 }
 
-fn one_message_sent_before_mc_trace(msg: Message) -> Vec<LogEntry> {
+fn one_message_sent_before_mc_trace(msg: Message, conf_type: McSystemConfiguringType) -> Vec<LogEntry> {
     let mut trace = two_nodes_started_trace();
-    trace.extend(vec![
-        LogEntry::LocalMessageReceived {
-            time: 0.0,
-            msg_id: "node1-process1-0".to_string(),
-            node: "node1".to_string(),
-            proc: "process1".to_string(),
-            msg: msg.clone(),
-        },
-        LogEntry::MessageSent {
-            time: 0.0,
-            msg_id: "0".to_string(),
-            src_node: "node1".to_string(),
-            src_proc: "process1".to_string(),
-            dst_node: "node2".to_string(),
-            dst_proc: "process2".to_string(),
-            msg,
-        },
-    ]);
+    match conf_type {
+        McSystemConfiguringType::DefaultSimulation => {
+            trace.extend(vec![
+                LogEntry::LocalMessageReceived {
+                    time: 0.0,
+                    msg_id: "node1-process1-0".to_string(),
+                    node: "node1".to_string(),
+                    proc: "process1".to_string(),
+                    msg: msg.clone(),
+                },
+                LogEntry::MessageSent {
+                    time: 0.0,
+                    msg_id: "0".to_string(),
+                    src_node: "node1".to_string(),
+                    src_proc: "process1".to_string(),
+                    dst_node: "node2".to_string(),
+                    dst_proc: "process2".to_string(),
+                    msg,
+                },
+            ]);
+        }
+        McSystemConfiguringType::McPreliminaryCallback => {
+            trace.extend(vec![
+                LogEntry::McLocalMessageReceived {
+                    proc: "process1".to_string(),
+                    msg: msg.clone(),
+                },
+                LogEntry::McMessageSent {
+                    src: "process1".to_string(),
+                    dst: "process2".to_string(),
+                    msg,
+                },
+            ]);
+        }
+    }
     trace
 }
 
+pub enum McSystemConfiguringType {
+    DefaultSimulation,
+    McPreliminaryCallback,
+}
```

Comment:
Предлагаю назвать так и перенести вверх:

```suggestion
pub enum SystemInitMethod {
    Simulation,
    PreliminaryCallback,
}
```